src/Controller/API/ProductController.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Controller\API;
+
+use App\Entity\Product;
+use App\Form\ProductType;
+use App\Repository\ProductRepository;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\OptimisticLockException;
+use Doctrine\ORM\ORMException;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+
+/**
+ * ProductController
+ * @Route("/api", name="api_")
+ */
+class ProductController extends AbstractController
+{
+    /**
+     * @var ProductRepository $productRepository
+     */
+    private $productRepository;
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * ProductController constructor.
+     * @param ProductRepository $productRepository
+     * @param EntityManagerInterface $em
+     */
+    public function __construct(ProductRepository $productRepository, EntityManagerInterface $em)
+    {
+        $this->productRepository = $productRepository;
+        $this->em = $em;
+    }
+
+    /** Show all Products.
+     * @Rest\Get("/product/all")
```

Comment:
wg konwencji, url api wystarczy ("/product") zamiast ("/product/all"). Jeśli nie ma podanego {id}, to wiadomo, że chodzi o listę produktów